### ISSUE TITLE
kubeflow stucks at juju bootstrap uk8s

### ISSUE THREAD

**Issue by rexlow** [canonical/microk8s#2398]
I am following this [guide](https://www.kubeflow.org/docs/distributions/microk8s/kubeflow-on-microk8s/#installing-and-enabling-kubeflow-using-microk8s) to run `kubeflow` in my Ubuntu 18.04 local environment.

The command was `KUBEFLOW_DEBUG=true microk8s enable kubeflow`

I've looked into the inspection report and it seems like juju is having trouble reaching the mongo server. It just forever loops in there.

Errors as follows:
```
2021-07-01 18:06:58 DEBUG juju.worker.apicaller connect.go:155 [84064c] failed to connect
2021-07-01 18:06:58 DEBUG juju.worker.dependency engine.go:598 "api-caller" manifold worker stopped: [84064c] "controller-0" cannot open api: unable to connect to API: dial tcp [::1]:17070: socket: address family not supported by protocol
2021-07-01 18:06:58 ERROR juju.worker.dependency engine.go:671 "api-caller" manifold worker returned unexpected error: [84064c] "controller-0" cannot open api: unable to connect to API: dial tcp [::1]:17070: socket: address family not supported by protocol
2021-07-01 18:06:58 DEBUG juju.worker.dependency engine.go:673 stack trace:
dial tcp [::1]:17070: socket: address family not supported by protocol
/home/jenkins/workspace/BuildJuju-amd64/_build/src/github.com/juju/juju/api/apiclient.go:716: 
/home/jenkins/workspace/BuildJuju-amd64/_build/src/github.com/juju/juju/api/apiclient.go:1136: 
/home/jenkins/workspace/BuildJuju-amd64/_build/src/github.com/juju/juju/api/apiclient.go:1111: unable to connect to API
/home/jenkins/workspace/BuildJuju-amd64/_build/src/github.com/juju/juju/api/apiclient.go:1007: 
/home/jenkins/workspace/BuildJuju-amd64/_build/src/github.com/juju/juju/api/apiclient.go:662: 
/home/jenkins/workspace/BuildJuju-amd64/_build/src/github.com/juju/juju/api/apiclient.go:239: 
/home/jenkins/workspace/BuildJuju-amd64/_build/src/github.com/juju/juju/worker/apicaller/connect.go:156: 
/home/jenkins/workspace/BuildJuju-amd64/_build/src/github.com/juju/juju/worker/apicaller/connect.go:211: 
/home/jenkins/workspace/BuildJuju-amd64/_build/src/github.com/juju/juju/worker/apicaller/manifold.go:97: [84064c] "controller-0" cannot open api
2021-07-01 18:07:13 DEBUG juju.cmd.jujud machine.go:1178 mongo is already initialized
2021-07-01 18:07:13 DEBUG juju.agent agent.go:925 potential mongo addresses: [localhost:37017]
2021-07-01 18:07:13 DEBUG juju.mongo open.go:160 mongodb connection failed, will retry: dial tcp 127.0.0.1:37017: connect: connection refused
2021-07-01 18:07:14 DEBUG juju.mongo open.go:160 mongodb connection failed, will retry: dial tcp 127.0.0.1:37017: connect: connection refused
2021-07-01 18:07:14 DEBUG juju.mongo open.go:160 mongodb connection failed, will retry: dial tcp 127.0.0.1:37017: connect: connection refused
2021-07-01 18:07:15 DEBUG juju.mongo open.go:160 mongodb connection failed, will retry: dial tcp 127.0.0.1:37017: connect: connection refused
2021-07-01 18:07:16 DEBUG juju.mongo open.go:160 mongodb connection failed, will retry: dial tcp 127.0.0.1:37017: connect: connection refused
```

If I try to execute again, another error would pop out
```
ERROR controller "uk8s" already exists
02:56:07 DEBUG cmd supercommand.go:537 error stack: 
/home/jenkins/workspace/BuildJuju-k8s-amd64/_build/src/github.com/juju/juju/environs/bootstrap/prepare.go:88: controller "uk8s" already exists
/home/jenkins/workspace/BuildJuju-k8s-amd64/_build/src/github.com/juju/juju/cmd/juju/commands/bootstrap.go:798: 

Command '('microk8s-juju.wrapper', '--debug', 'bootstrap', 'microk8s', 'uk8s')' returned non-zero exit status 1.
```

Here I have attached the log files [inspection-report-20210702_025015.tar.gz](https://github.com/ubuntu/microk8s/files/6750246/inspection-report-20210702_025015.tar.gz)

**Comment by rexlow** [comment]
Found an exact issue[here](https://github.com/kubeflow/kubeflow/issues/5034) in the kubeflow repo

**Comment by DomFleischmann** [comment]
can you try and first disable the addon before retrying?
`microk8s disable kubeflow`

Else check if you can delete the controller namespace directly
`microk8s kubectl delete namespace controller-uk8s`

Also it would be good to know how much resources your environment has as that might also be an issue.

**Comment by rexlow** [comment]
@DomFleischmann I did that and tried to reinstall kubeflow from scratch. Same problem happens again and I managed to find the root cause. It was `juju` having trouble connecting with mongodb. It's stucked in the dial up loop.

I ended up clean install a fresh Ubuntu 20.04 and now everything works. But the jupyter page is not rendering properly.

For reference, my host PC is running with 48GB RAM, 1TB SSD and 2x Nvidia RTX 2060.
